python_modules/jupedsim/tests/test_distributions.py

```diff
@@ -0,0 +1,291 @@
+
+
+def test_catch_wrong_inputs_for_wrong_length_center_point():
+    polygon = distributions.shply.Polygon([(0, 0), (10, 0), (10, 10), (0, 10)])
+    center_point = 5
+    # center_point is no tuple
+    circle_segment_radii, numbers_of_agents = [(1, 3)], [5]
+    with pytest.raises(distributions.IncorrectParameterError):
+        distributions.__catch_wrong_inputs(polygon=polygon, center_point=center_point,
+                                           circle_segment_radii=circle_segment_radii, fill_parameters=numbers_of_agents)
+    center_point = [0]
+    # center_point only contains 1 number
+    with pytest.raises(distributions.IncorrectParameterError):
+        distributions.__catch_wrong_inputs(polygon=polygon, center_point=center_point,
+                                           circle_segment_radii=circle_segment_radii, fill_parameters=numbers_of_agents)
+
+
+def test_catch_wrong_inputs_for_not_enough_fill_parameter():
+    polygon = distributions.shply.Polygon([(0, 0), (10, 0), (10, 10), (0, 10)])
+    center_point = (0, 0)
+    circle_segment_radii, numbers_of_agents = [(1, 3), (3, 5)], [5]
+    # two circle segments given only one fill parameter
+    with pytest.raises(distributions.IncorrectParameterError):
+        distributions.__catch_wrong_inputs(polygon=polygon, center_point=center_point,
+                                           circle_segment_radii=circle_segment_radii, fill_parameters=numbers_of_agents)
+
+
+def test_catch_wrong_inputs_for_negativ_radius():
+    polygon = distributions.shply.Polygon([(0, 0), (10, 0), (10, 10), (0, 10)])
+    center_point, numbers_of_agents = (0, 0), [5, 5]
+    circle_segment_radii = [(-1, 3), (3, 5)]
+    with pytest.raises(distributions.NegativeValueError):
+        distributions.__catch_wrong_inputs(polygon=polygon, center_point=center_point,
+                                           circle_segment_radii=circle_segment_radii, fill_parameters=numbers_of_agents)
+
+
+def test_catch_wrong_inputs_for_Overlapping_Circles():
+    polygon = distributions.shply.Polygon([(0, 0), (10, 0), (10, 10), (0, 10)])
+    center_point, numbers_of_agents = (0, 0), [5, 5]
+    circle_segment_radii = [(1, 3), (2, 5)]
+    with pytest.raises(distributions.OverlappingCirclesError):
+        distributions.__catch_wrong_inputs(polygon=polygon, center_point=center_point,
+                                           circle_segment_radii=circle_segment_radii, fill_parameters=numbers_of_agents)
+    circle_segment_radii, numbers_of_agents = [(0, 2), (3, 5), (2, 4)], [5, 5, 5]
+    with pytest.raises(distributions.OverlappingCirclesError):
+        distributions.__catch_wrong_inputs(polygon=polygon, center_point=center_point,
+                                           circle_segment_radii=circle_segment_radii, fill_parameters=numbers_of_agents)
+
+
+def test_catch_wrong_inputs_for_minimum_radius_bigger_than_maximum_radius():
```

Comment:
This method could be part of the upper test method by parametrizing as well

Reply:
done